grammar.js

```diff
@@ -908,6 +980,23 @@ module.exports = grammar(C, {
 
     lambda_default_capture: $ => choice('=', '&'),
 
+    _fold_operator: $ => choice(...FOLD_OPERATORS),
+    _binary_fold_operator: $ => choice(...FOLD_OPERATORS.map(operator => seq(operator, '...', operator))),
+
+    _unary_right_fold: $ => seq('...', $._fold_operator, $._expression),
```

Comment:
For all three of these folds, can you add `field()` labels to the `...`, operator, and expression like normal `binary_expression`s do?

For binary folds, I wonder whether we label the entire "middle part" (`op .. op`) as one, or the individual pieces...not sure I have a strong opinion on that one.